app/views/declaration/additional_fiscal_references.scala.html

```diff
@@ -43,26 +43,57 @@
 
 @(mode: Mode, itemId:String, form: Form[AdditionalFiscalReference], additionalReferences: Seq[AdditionalFiscalReference] = Seq.empty)(implicit request: Request[_], messages: Messages)
 
+@referenceTable = {
+    @if(additionalReferences.nonEmpty){
+        <div class="field-group mb-2">
+            <table id="removable_elements">
+
+                <thead>
+                    <tr>
+                        <th id="@messages("declaration.additionalFiscalReferences.numbers.header")-heading" colspan="2">@messages("declaration.additionalFiscalReferences.numbers.header")</th>
+                    </tr>
+                </thead>
+
+                <tbody>
+                @for((elem, index) <- additionalReferences.map(_.asString).zipWithIndex) {
+                    <tr id="removable_elements__row@index">
+                        <td id="removable_elements-row@index-label">@{elem}</td>
+                        <td id="removable_elements-row@index-remove_button">
+                            <input
+                                type="submit"
+                                class="button--secondary"
+                                formmethod="post"
+                                formaction="@routes.AdditionalFiscalReferencesController.removeReference(mode, itemId, elem)"
+                                value="@messages("site.remove")"
+                            >
+                        </td>
+                    </tr>
+                }
+                </tbody>
+            </table>
+        </div>
+    }
+}
+
 @main_template(
     title = messages("declaration.additionalFiscalReferences.title")
 ) {
 
-    @components.back_link(routes.FiscalInformationController.displayPage(mode, itemId), mode)
+    @components.back_link(routes.FiscalInformationController.displayPage(mode, itemId))
 
     @components.error_summary(form.errors)
 
     @components.section_header(messages("declaration.fiscalInformation.header"))
 
     @components.page_title(Some(messages("declaration.additionalFiscalReferences.title")))
 
-    @components.fields.removable_elements_table(
-        title = Some(messages("declaration.additionalFiscalReferences.numbers.header")),
-        elements = additionalReferences.map(_.asString).map(ref => LabelledValue(ref))
-    )
-
     @helper.form(routes.AdditionalFiscalReferencesController.saveReferences(mode, itemId), 'autoComplete -> "off") {
         @helper.CSRF.formField
 
+
+        @referenceTable
```

Comment:
Seems unnecessary to pull this out as it’s not reused

Reply:
This is more about to keep high level point of view whole page - we have components earlier - and later - this lot of low level tags just breaks abstraction level here

This is why it why i extract those just as variable in same view